### ISSUE TITLE
The import com.google cannot be resolved

### ISSUE THREAD

**Issue by pixxio** [phonegap/phonegap-plugin-barcodescanner#286]
After installing the newest version (6.0.1) for Android i got the error "The import com.google cannot be resolved" in BarcodeScanner.java (In this line: "import com.google.zxing.client.android.Intents;").
Does anybody know how to solve this problem?
I checked several other issue threads but none of them helped me.
I am using cordova 6.2.0.

Any help is highly appreciated, thanks!

**Comment by detectivemerry** [comment]
Also having this problem

**Comment by pixxio** [comment]
@daserge 
You don't have any clue, do you?

Thanks in advance,
pixxio

**Comment by jcesarmobile** [comment]
Where do you see this error?

**Comment by pixxio** [comment]
In eclipse.
For example when tell eclipse to build a .apk file.


Am 04.08.2016 um 01:14 schrieb jcesarmobile <notifications@github.com<mailto:notifications@github.com>>


Where do you see this error?

-

**Comment by jcesarmobile** [comment]
The plugin is no longer compatible with Eclipse.
With Cordova you shouldn't use any IDE, just the CLI, but in case you wan't to use one, it should be Android Studio as Google deprecated Eclipse long time ago.

The reason for your error is the plugin now uses a .aar instead of a .jar for zxing (the library used to read bar codes)
https://github.com/phonegap/phonegap-plugin-barcodescanner/blob/master/plugin.xml#L95

**Comment by amiral84** [comment]
@jcesarmobile so what's the fix? I know the reason but I'm not able to use .aar file even though I have added it in my libs folder. I tried to add the `classes.jar` in libs folder and it found all zxings imports but the problem is when I try to open the barcode scanner I get following error:

```
E/AndroidRuntime: FATAL EXCEPTION: main
                                                                 Process: com.test.app, PID: 26675
                                                                 java.lang.NoClassDefFoundError: Failed resolution of: Lbarcodescanner/xservices/nl/barcodescanner/R$layout;
                                                                     at com.google.zxing.client.android.CaptureActivity.onCreate(CaptureActivity.java:150)
                                                                     at android.app.Activity.performCreate(Activity.java:5977)
                                                                     at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105)
                                                                     at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2258)
                                                                     at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2365)
                                                                     at android.app.ActivityThread.access$800(ActivityThread.java:148)
                                                                     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1283)
                                                                     at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                     at android.os.Looper.loop(Looper.java:135)
                                                                     at android.app.ActivityThread.main(ActivityThread.java:5272)
                                                                     at java.lang.reflect.Method.invoke(Native Method)
                                                                     at java.lang.reflect.Method.invoke(Method.java:372)
                                                                     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:909)
                                                                     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:704)
                                                                  Caused by: java.lang.ClassNotFoundException: Didn't find class "barcodescanner.xservices.nl.barcodescanner.R$layout" on path: DexPathList[[zip file "/data/app/com.test.app-1/base.apk"],nativeLibraryDirectories=[/vendor/lib, /system/lib]]
                                                                     at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
                                                                     at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
                                                                     at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
                                                                     at com.google.zxing.client.android.CaptureActivity.onCreate(CaptureActivity.java:150) 
                                                                     at android.app.Activity.performCreate(Activity.java:5977) 
                                                                     at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105) 
                                                                     at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2258) 
                                                                     at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2365) 
                                                                     at android.app.ActivityThread.access$800(ActivityThread.java:148) 
                                                                     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1283) 
                                                                     at android.os.Handler.dispatchMessage(Handler.java:102) 
                                                                     at android.os.Looper.loop(Looper.java:135) 
                                                                     at android.app.ActivityThread.main(ActivityThread.java:5272) 
                                                                     at java.lang.reflect.Method.invoke(Native Method) 
                                                                     at java.lang.reflect.Method.invoke(Method.java:372) 
                                                                     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:909) 
                                                                     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:704) 
                                                                 	Suppressed: java.lang.ClassNotFoundException: barcodescanner.xservices.nl.barcodescanner.R$layout
                                                                     at java.lang.Class.classForName(Native Method)
                                                                     at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
                                                                     at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
                                                                     at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
                                                                     		... 15 more
                                                                  Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack available
```

The problem here is obvious, class not found. But does anybody knows the solution for this?

**Comment by jcesarmobile** [comment]
There is no fix, you have to use Cordova CLI or Android Studio to run the app. The plugin now uses an AAR file instead of the old jar, and AAR files are not compatible with Eclipse.
If you are using Cordova CLI or Android Studio you shouldn't see this error. If you are using them and still see the error message create a new issue providing all the information possible as Cordova CLI version, cordova-android platform version, etc.

**Comment by amiral84** [comment]
@jcesarmobile I know it is not compatible, you have stated that before and I would not have asked about this issue if I would not have used Android Studio. So to make this clear, I use Android Studio and the project won't compile because it can't find zxings classes from that .aar file. 

But I will make a new issue, even though it is the same problem of this issue. I have Ionic project and I use Cordova CLI to run cordova commands, etc.

**Comment by jcesarmobile** [comment]
It seems the same problem but it isn't, the other problem was because he was using Eclipse and you are using Android Studio, so you have to create a new issue for that.

On a totally new Cordova app, just adding Android platform and the phonegap-plugin-barcodescanner I can run the project and it compiles on both the CLI with `cordova run android --device` or the run button from Android Studio.
You said that you added the .aar in your lib folder, you don't have to do it, it's done on plugin install.

That's why you have to open a new issue, because you have to provide all the possible information, what you did, how you did it, CLI and platform versions you are using, etc.

**Comment by amiral84** [comment]
Finally resolved it. I had to do many things for my project. Here are some of the things that I had to change in my project:

1. Configure Gradle to use 2.14.0
2. Import .aar file as new module (File -> New -> New Module -> Import .JAR/.AAR file) and add this module as dependency to my project
3. Change `compileSdkVersion` and `buildToolsVersion` to version 19 and '21.1.2'
4. Clean / Rebuild the project
5. Run

I had to change my IDE for one sake of one file (.aar) :(, because IntelliJ IDEA couldn't for some reason import .aar file as module even though it would allow so.

Hopefully those steps helps others.
